ddpui/management/commands/dumpconfig.py

```diff
@@ -0,0 +1,142 @@
+from ddpui.utils.custom_logger import CustomLogger
+from ddpui.utils.secretsmanager import retrieve_warehouse_credentials
+from ddpui.ddpairbyte import airbyte_service
+
+logger = CustomLogger("ddpui")
+load_dotenv()
+
+class Command(BaseCommand):
+    """
+    This script dumps the configuration of an org to file
+    The configuration can be loaded into a fresh Dalgo instance
+    """
+
+    help = "Dumps an organization's config"
+
+    def __init__(self):
+        super().__init__()
+        self.output_config = {}
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--org",
+            required=True,
+            default="all",
+            help="The slug for the org",
+        )
+        parser.add_argument("--output", required=False, help="The output file")
+
+    def dump_warehouse(self, org: Org):
+        """dumps warehouse information to the output config"""
+        warehouse = OrgWarehouse.objects.filter(org=org).first()
+        if warehouse is None:
+            logger.error(f"No warehouse found for {org.slug}")
+            return None
+        retval = {}
+        retval["wtype"] = warehouse.wtype
+        retval["airbyte_destination_id"] = warehouse.airbyte_destination_id
+        retval["bq_location"] = warehouse.bq_location
+        retval["credentials"] = retrieve_warehouse_credentials(warehouse)
+        retval["destination"] = airbyte_service.get_destination(
+            org.airbyte_workspace_id, warehouse.airbyte_destination_id
+        )
+        return retval
+
+    def dump_sources(self, org: Org):
+        """dumps all sources for an org"""
+        source_definitions = airbyte_service.get_source_definitions(
+            org.airbyte_workspace_id
+        )["sourceDefinitions"]
+        sources = airbyte_service.get_sources(org.airbyte_workspace_id)["sources"]
+        for source in sources:
+            for sdef in source_definitions:
+                if sdef["sourceDefinitionId"] == source["sourceDefinitionId"]:
+                    source["sourceDefinitionName"] = sdef["name"]
+                    break
+        return sources
+
+    def dump_connections(self, org: Org):
+        """dumps airbyte connection info"""
+        connections = airbyte_service.get_connections(org.airbyte_workspace_id)["connections"]
+        formatted_connections = []
+        for connection in connections:
+            connection["destination"] = airbyte_service.get_destination(
+                org.airbyte_workspace_id, connection["destinationId"])
+            connection["source"] = airbyte_service.get_source(
+                org.airbyte_workspace_id, connection["sourceId"])
+            streams = connection["syncCatalog"]["streams"]
+            formatted_streams = []
+            for stream in streams:
+                formatted_stream = {
+                    "name": stream["stream"]["name"],
+                    "supportsIncremental": True,
```

Comment:
how come this is hardcoded to `True`?

Reply:
oh we don't actually need to dump supportsIncremental, removed it

Reply:
ok ... can you explain what it is though, why it is there, why we don't need it

Reply:
supportsIncremental is to check if the source supports incremental sync or not. it is not needed to create a connection because it takes it by default depending on the source. so we do not need to dump it